tests/tracer/test_correlation_log_context.py

```diff
@@ -0,0 +1,136 @@
+    def test_get_log_correlation_context(self, global_config):
+        """Ensure expected DDLogRecord is generated via get_correlation_log_record."""
+        with tracer.trace("test-span-1") as span1:
+            dd_log_record = tracer.get_log_correlation_context()
+        assert dd_log_record.span_id == str(span1.span_id)
+        assert dd_log_record.trace_id == str(span1.trace_id)
+        assert dd_log_record.service == "test-service"
+        assert dd_log_record.env == "test-env"
+        assert dd_log_record.version == "test-version"
+
+        test_tracer = Tracer()
+        with test_tracer.trace("test-span-2") as span2:
+            dd_log_record = test_tracer.get_log_correlation_context()
+        assert dd_log_record.span_id == str(span2.span_id)
+        assert dd_log_record.trace_id == str(span2.trace_id)
+        assert dd_log_record.service == "test-service"
+        assert dd_log_record.env == "test-env"
+        assert dd_log_record.version == "test-version"
+
+    def test_get_log_correlation_context_opentracer(self, global_config):
+        """Ensure expected DDLogRecord generated via get_correlation_log_record with an opentracing Tracer."""
+        ot_tracer = OT_Tracer()
+        with ot_tracer.start_active_span("operation") as scope:
+            dd_span = scope._span._dd_span
+            dd_log_record = ot_tracer.get_log_correlation_context()
+        assert dd_log_record.span_id == str(dd_span.span_id)
+        assert dd_log_record.trace_id == str(dd_span.trace_id)
+        assert dd_log_record.service == "test-service"
+        assert dd_log_record.env == "test-env"
+        assert dd_log_record.version == "test-version"
+
+    def test_get_log_correlation_context_no_active_span(self):
+        """Ensure empty DDLogRecord generated if no active span."""
+        dd_log_record = tracer.get_log_correlation_context()
+        assert dd_log_record.span_id == "0"
+        assert dd_log_record.trace_id == "0"
+        assert dd_log_record.service == ""
+        assert dd_log_record.env == ""
+        assert dd_log_record.version == ""
+
+    def test_get_log_correlation_context_disabled_tracer(self):
+        """Ensure get_correlation_log_record returns None if tracer is disabled."""
+        tracer = Tracer()
+        tracer.enabled = False
+        with tracer.trace("test-span"):
+            dd_log_record = tracer.get_log_correlation_context()
+        assert dd_log_record.span_id == "0"
+        assert dd_log_record.trace_id == "0"
+        assert dd_log_record.service == ""
+        assert dd_log_record.env == ""
+        assert dd_log_record.version == ""
+
+    def test_custom_logging_injection(self):
+        """Ensure custom log injection via get_correlation_log_record returns proper active span information."""
+        capture_log = structlog.testing.LogCapture()
+        structlog.configure(processors=[tracer_injection, capture_log, structlog.processors.JSONRenderer()])
```

Comment:
does `capture_log` need to come after `JSONRenderer`?

can we test that the JSON produced is correct?

Reply:
Nope, according to their docs `JSONRenderer` is usually the last processor in the chain, and trying `capture_log` after `JSONRenderer` leads to TypeErrors with the `event_dict` being mutated by the `JSONRenderer` and unable to support item assignment. The `JSONRenderer()` by default uses `json.dumps()` to format the string, and since our new API will always output a Dictionary with string keys and string values, we shouldn't need to worry about this not being serializable.